### PR TITLE
SOL-95364: 400 Invalid Queue Endpoint return code

### PR DESCRIPTION
**Pull request by TrentDaniel**
Added test to verify that the CCSMP subcode 164 is mapped correctly and is passed to the application through an error when a persistent receiver tries to provision and bind to a non durable queue but uses the name of a pre-existing durable queue.

### CODE REVIEW

**Comment by oodigie** [comment]
@TrentDaniel, it looks like you have some failing unit tests in this PR - https://github.com/SolaceDev/pubsubplus-go-client/actions/runs/10925723303/job/30327880565?pr=63

File: test/persistent_receiver_test.go [hunk 1]
```diff
@@ -148,6 +150,35 @@ var _ = Describe("PersistentReceiver", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(fmt.Sprint(receiver)).To(ContainSubstring(fmt.Sprintf("%p", receiver)))
 		})
+		Context("with a connected messaging service and specially formatted queue", func() {
+            slog.SetLogLoggerLevel(slog.LevelDebug)
+            const basicQueueName = "test_invalid_durability_on_bind_raises_exception"
+            modifiedQueueName := "#P2P/QTMP/v:solbroker/" + basicQueueName
+
+			BeforeEach(func() {
+                fmt.Println("Starting test")
```
Comment (oodigie) [inline test/persistent_receiver_test.go:159]: @TrentDaniel, I don't think we should have this print statement here.

File: test/persistent_receiver_test.go [hunk 2]
```diff
@@ -148,6 +150,35 @@ var _ = Describe("PersistentReceiver", func() {
...
+			AfterEach(func() {
+				if messagingService.IsConnected() {
+					helpers.DisconnectMessagingService(messagingService)
+				}
+				helpers.DeleteQueue(url.QueryEscape(modifiedQueueName))
+			})
+
+            Describe("Invalid Queue or Topic Endpoint", func() {
+                It("can return an error when configured to bind to a non-durable queue but attempts to bind to a durable queue.", func() {
+                    fmt.Println("Got to It section")
```
Comment (oodigie) [inline test/persistent_receiver_test.go:173]: @TrentDaniel, I don't think we should have this print statement here too.

File: test/persistent_receiver_test.go [hunk 3]
```diff
@@ -148,6 +150,35 @@ var _ = Describe("PersistentReceiver", func() {
...
+                    receiver := helpers.NewPersistentReceiver(messagingService, resource.QueueNonDurableExclusive(basicQueueName))
+                    err := receiver.Start()
+                    Expect(err).ShouldNot(BeNil())
+                    fmt.Printf("err is: %s", err.Error())
```
Comment (oodigie) [inline test/persistent_receiver_test.go:177]: @TrentDaniel, this one too.

Comment (TrentDaniel) [reply test/persistent_receiver_test.go]: Fixed in 33a49baf262fe9ed0b7b41fd899ffe3e42f8ab56, please resolve this conversation if you agree.

File: test/persistent_receiver_test.go (hunk 1 above)
Comment (TrentDaniel) [reply test/persistent_receiver_test.go]: Fixed in https://github.com/SolaceDev/pubsubplus-go-client/commit/33a49baf262fe9ed0b7b41fd899ffe3e42f8ab56, please resolve this conversation if you agree.

**Comment by TrentDaniel** [comment]
> @TrentDaniel, it looks like you have some failing unit tests in this PR - https://github.com/SolaceDev/pubsubplus-go-client/actions/runs/10925723303/job/30327880565?pr=63

@oodigie , this unit test failure seems to be a timeout during a test that was not changed by this branch. This branch only added a test, and did not make changes to the implementation or preexisting test that failed before. Have you seen a problem like this in the API before? If there is not already a ticket open to address this, then we should open one now. I was unable to reproduce the issue locally, although I only ran the test a few times.

**Comment by oodigie** [comment]
I think if we should be fine if you rerun the Git-actions and it resolves. And also if you cannot reproduce the test failure locally. It probably might have been something with the node in Git-actions or some rare race condition test.

File: test/persistent_receiver_test.go [hunk 4]
```diff
@@ -148,6 +150,32 @@ var _ = Describe("PersistentReceiver", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(fmt.Sprint(receiver)).To(ContainSubstring(fmt.Sprintf("%p", receiver)))
 		})
+		Context("with a connected messaging service and specially formatted queue", func() {
+            slog.SetLogLoggerLevel(slog.LevelDebug)
```
Comment (oodigie) [inline test/persistent_receiver_test.go:154]: @TrentDaniel, please any reason why you included this ?

**Review by oodigie — CHANGES_REQUESTED** [review]

File: test/persistent_receiver_test.go (hunk 4 above)
Comment (TrentDaniel) [reply test/persistent_receiver_test.go:154]: I included that for debugging purposes, I will turn it off now, good catch

File: test/persistent_receiver_test.go [hunk 5]
```diff
@@ -148,6 +149,31 @@ var _ = Describe("PersistentReceiver", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(fmt.Sprint(receiver)).To(ContainSubstring(fmt.Sprintf("%p", receiver)))
 		})
+		Context("with a connected messaging service and specially formatted queue", func() {
+            const basicQueueName = "test_invalid_durability_on_bind_raises_exception"
+            modifiedQueueName := "#P2P/QTMP/v:solbroker/" + basicQueueName
```
Comment (cjwmorgan-sol) [inline test/persistent_receiver_test.go]: Does this only work as solbroker is the default hostname? Is there a way to pull the docker hostname?
Does this break again remote broker hostname?

**Review by cjwmorgan-sol — CHANGES_REQUESTED** [review]
The use of a hardcoded solbroker in the v-rid name of the temporary endpoint name should be data driven. There is PUBSUB_HOSTNAME envar/testContext injection. The name should be message.hostname or testcontainer.broker_hostname driven not hardcoded.

File: test/persistent_receiver_test.go (hunk 5 above)
Comment (oodigie) [reply test/persistent_receiver_test.go]: I agree with @cjwmorgan-sol. Please is there a way to retrieve the hostname dynamically?

**Comment by TrentDaniel** [comment]
> The use of a hardcoded solbroker in the v-rid name of the temporary endpoint name should be data driven. There is PUBSUB_HOSTNAME envar/testContext injection. The name should be message.hostname or testcontainer.broker_hostname driven not hardcoded.

When I run my poc solution locally, the API will create a non-durable queue on the broker with the string solbroker instead of localhost or whatever else I try to use for the queue. Either this is hardcoded into the API or its hardcoded in the broker, which could be possible since this aspect of the queue name is not configurable by the customer. @cjwmorgan-sol , if there are no other changes needed, please approve and merge this PR. If you have questions about the queue name, we can go over it tomorrow.

**Comment by cjwmorgan-sol** [comment]
> Either this is hardcoded into the API or its hardcoded in the broker,
This is configured in the broker using cli commands:
```
enable
configure
hostname <name>
```
When using containers it ends up as the container hostname from the docker compose.
I took a look into the testcontext. The testcontext.Messaging().Hostname is the messaging connection (or PUBSUB_HOST) not the broker hostname (PUBSUB_HOSTNAME). So as you have determined when set to localhost this will not work. There is no testcontext.TestContainer configuration that is accessible outside of the testcontext module. An accessor could be added, but there is another way as well using receiverInfo and binding a flow with a non-durable queue.

I would add the only other use of `#P2P/QUE/v:solbroker` in the tests attempts to subscription to a queue as a topic where `#P2P/QUE/` prefix is likely enough. Ie the v:solbroker does not matter. 

The usage here for the non-durable name queue the `#P2P/QTMP/v:<vrid>` must match the configured hostname. Leading to issues as the target broker hostname is not always solbroker. 

I think the best approach here is to bind to a non-durable queue endpoint then get the name and parse out the vrid from the pattern `#P2P/QTMP/v:([a-zA-Z0-9]+)/.*` (note please double check the regex) if that parsing does not work default to solbroker though ideally default to testcontext.TestContainer.BrokerHostName then when empty default to solbroker.

**Review by oodigie — APPROVED** [review]
LGTM!

**Pull request merged**